mvcc/key_index.go

```diff
@@ -187,6 +187,19 @@ func (ki *keyIndex) compact(atRev int64, available map[revision]struct{}) {
 		plog.Panicf("store.keyindex: unexpected compact on empty keyIndex %s", string(ki.key))
 	}
 
+	ki.doCompact(atRev, available, true)
+}
+
+// keep finds the revision to be kept if compact is called at given atRev.
+func (ki *keyIndex) keep(atRev int64, available map[revision]struct{}) {
+	if ki.isEmpty() {
+		return
+	}
+
+	ki.doCompact(atRev, available, false)
+}
+
+func (ki *keyIndex) doCompact(atRev int64, available map[revision]struct{}, doModify bool) {
```

Comment:
it's not good practice to have boolean arguments; possibly have
`func (ki *keyIndex) doCompact(atRev int64, available map[revision]struct{}) (genIdx int,  revIndex int)` and handle the modification after calling doCompact